placeholder-material/src/main/java/com/google/accompanist/placeholder/material/Placeholder.kt

```diff
@@ -0,0 +1,100 @@
+ */
+
+package com.google.accompanist.placeholder.material
+
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.contentColorFor
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.compositeOver
+import com.google.accompanist.placeholder.PlaceholderHighlight
+import com.google.accompanist.placeholder.placeholder
+
+/**
+ * Contains default values used by [Modifier.placeholder] and [PlaceholderHighlight].
+ */
+object PlaceholderDefaults {
+    /**
+     * Returns the value used as the the `color` parameter value on [Modifier.placeholder].
+     *
+     * @param backgroundColor The current background color of the layout. Defaults to
+     * `MaterialTheme.colors.surface`.
+     */
+    @Composable
+    fun color(
+        backgroundColor: Color = MaterialTheme.colors.surface
+    ): Color {
+        return contentColorFor(backgroundColor).copy(alpha = 0.1f).compositeOver(backgroundColor)
+    }
+
+    /**
+     * Returns the value used as the the `highlightColor` parameter value of
+     * [PlaceholderHighlight.Companion.fade].
+     *
+     * @param backgroundColor The current background color of the layout. Defaults to
+     * `MaterialTheme.colors.surface`.
+     */
+    @Composable
+    fun fadeHighlightColor(
+        backgroundColor: Color = MaterialTheme.colors.surface
+    ): Color {
+        return backgroundColor.copy(alpha = 0.3f)
+    }
+
+    /**
+     * Returns the value used as the the `highlightColor` parameter value of
+     * [PlaceholderHighlight.Companion.shimmer].
+     *
+     * @param backgroundColor The current background color of the layout. Defaults to
+     * `MaterialTheme.colors.surface`.
+     */
+    @Composable
+    fun shimmerHighlightColor(
+        backgroundColor: Color = MaterialTheme.colors.surface
+    ): Color {
+        return backgroundColor.copy(alpha = 0.75f)
+    }
+}
+
+/**
+ * Draws some skeleton UI which is typically used whilst content is 'loading'.
+ *
+ * To customize the color and shape of the placeholder, you can use the foundation version of
+ * [Modifier.placeholder], along with the values provided bu [PlaceholderDefaults].
+ *
+ * You can provide a [PlaceholderHighlight] which runs an highlight animation on the placeholder.
+ * The [shimmer] and [fade] implementations are provided for easy usage.
+ *
+ * @sample com.google.accompanist.sample.placeholder.PlaceholderFadeSample
+ * @sample com.google.accompanist.sample.placeholder.PlaceholderShimmerSample
+ *
+ * @param visible whether the placeholder should be visible or not.
+ * @param highlight optional highlight animation.
+ */
+fun Modifier.placeholder(
+    visible: Boolean,
+    highlight: PlaceholderHighlight? = null,
+): Modifier = composed {
+    Modifier.placeholder(
+        visible = visible,
+        color = PlaceholderDefaults.color(),
```

Comment:
Expose this as a param

Reply:
I disagree with needing to use the non-material version to customize anything, I'd prefer it can still be customized but offers material defaults.

Reply:
Can't be done. Modifiers shouldn't be composable and there's no way to use a composable value here (see the internal chat).